org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/JavadocCompletionProposal.java

```diff
@@ -0,0 +1,268 @@
+			throw new IllegalArgumentException("Compilation unit must not be null"); //$NON-NLS-1$
+		}
+		List<CompletionItem> result = new ArrayList<>();
+		IDocument d = JsonRpcHelpers.toDocument(cu.getBuffer());
+		if (offset < 0 || d.getLength() == 0) {
+			return result;
+		}
+		try {
+			int p = (offset == d.getLength() ? offset - 1 : offset);
+			IRegion line = d.getLineInformationOfOffset(p);
+			String lineStr = d.get(line.getOffset(), line.getLength()).trim();
+			if (!lineStr.startsWith("/**")) {
+				return result;
+			}
+			if (!hasEndJavadoc(d, offset)) {
+				return result;
+			}
+			String text = collector.getContext().getToken() == null ? "" : new String(collector.getContext().getToken());
+			StringBuilder buf = new StringBuilder(text);
+			IRegion prefix = findPrefixRange(d, line);
+			String indentation = d.get(prefix.getOffset(), prefix.getLength());
+			int lengthToAdd = Math.min(offset - prefix.getOffset(), prefix.getLength());
+			buf.append(indentation.substring(0, lengthToAdd));
+			String lineDelimiter = TextUtilities.getDefaultLineDelimiter(d);
+			ICompilationUnit unit = cu;
+			try {
+				unit.reconcile(ICompilationUnit.NO_AST, false, null, null);
+				String string = createJavaDocTags(d, offset, indentation, lineDelimiter, unit);
+				if (string != null && !string.trim().equals("*")) {
+					buf.append(string);
+				} else {
+					return result;
+				}
+				int nextNonWS = findEndOfWhiteSpace(d, offset, d.getLength());
+				if (!Character.isWhitespace(d.getChar(nextNonWS))) {
+					buf.append(lineDelimiter);
+				}
+			} catch (CoreException e) {
+				// ignore
+			}
+			final CompletionItem ci = new CompletionItem();
+			Range range = JDTUtils.toRange(unit, offset, 0);
+			ci.setTextEdit(new TextEdit(range, rtrim(buf.toString())));
+			ci.setFilterText(JAVA_DOC_COMMENT);
+			ci.setLabel(JAVA_DOC_COMMENT);
+			ci.setSortText(SortTextHelper.convertRelevance(0));
+			ci.setKind(CompletionItemKind.Snippet);
+			ci.setInsertTextFormat(InsertTextFormat.PlainText);
+			String documentation = prepareDocumentation(buf.toString(), lineDelimiter);
+			ci.setDocumentation(documentation);
+			Map<String, String> data = new HashMap<>(3);
+			data.put(CompletionResolveHandler.DATA_FIELD_URI, JDTUtils.toURI(cu));
+			data.put(CompletionResolveHandler.DATA_FIELD_REQUEST_ID, "0");
+			data.put(CompletionResolveHandler.DATA_FIELD_PROPOSAL_ID, "0");
+			ci.setData(data);
+			result.add(ci);
+		} catch (BadLocationException excp) {
+			// stop work
+		}
+		return result;
+	}
+
+	private String prepareDocumentation(String text, String lineDelimiter) {
+		String[] lines = text.split(lineDelimiter);
+		StringBuilder buf = new StringBuilder();
+		for (String line : lines) {
+			buf.append(rtrim(line));
+			buf.append(lineDelimiter);
+		}
+		String result = buf.toString();
+		if (result.indexOf(lineDelimiter) == 0) {
+			return result.replaceFirst(lineDelimiter, "");
+		}
+		return result;
+	}
+
+	private String rtrim(String text) {
```

Comment:
what is implemented is a ltrim, i.e. whitespaces on the left are removed. We want to remove the trailing spaces, thos on the right